File: sources/com/s8/pkgs/ui/carbide/breadcrumbs/BreadcrumbsItem.java
package com.s8.pkgs.ui.carbide.breadcrumbs;

import com.s8.api.web.S8WebFront;
import com.s8.api.web.S8WebObject;
import com.s8.pkgs.ui.carbide.WebSources;


/**
 * 
 * @author pierreconvert
 *
 */
public class BreadcrumbsItem extends S8WebObject {

	
	public final static String JS_TYPENAME = "/breadcrumbs/BreadcrumbsItem";
	
	/**
	 * 
	 * @param branch
	 */
	public BreadcrumbsItem(S8WebFront branch) {
		super(branch, WebSources.ROOT_WEBPATH + JS_TYPENAME);
	}
	
	
	public BreadcrumbsItem(S8WebFront branch, String content) {
		super(branch, WebSources.ROOT_WEBPATH + JS_TYPENAME);
		setContent(content);
	}
	
	
	
	/**
	 * 
	 * @param name
	 */
	public void setContent(String content) {
		vertex.outbound().setStringUTF8Field("content", content);
	}
	
	
}

File: web-sources/S8-pkgs-ui-carbide/breadcrumbs/BreadcrumbsItem.js


import { S8WebFront } from "/S8-pkgs-ui-carbide/S8WebFront.js";
import { S8Object } from "/S8-api/S8Object.js";




/**
 * 
 */
S8WebFront.CSS_import('/S8-pkgs-ui-carbide/breadcrumbs/Breadcrumbs.css');


/**
 * 
 */
export class BreadcrumbsItem extends S8Object {

    /** @type {HTMLDivElement} */
    wrapperNode;

    
    /**
     * 
     */
    constructor(){
        super();
        this.wrapperNode = document.createElement("div");
        this.wrapperNode.classList.add("breadcrumbs-item");
    }


    /**
     * 
     * @returns 
     */
    getEnvelope(){
        return this.wrapperNode;
    }


    /** @param {string} content */
    S8_set_content(content){
       this.wrapperNode.innerText = content;
    }


    /** */
    S8_render(){}


    /** */
    S8_dispose(){}
}